src/main/kotlin/step2/Calculator.kt

```diff
@@ -0,0 +1,27 @@
+package step2
+
+class Calculator {
+    private var total = 0
+
+    fun calculate(expression: String): Int {
+        val elements = expression.split(" ").groupBy { isNumeric(it) }
+        val numbers = elements[true]?.map { it.toInt() }
+        val operators = elements[false]!!
+
+        if (numbers.isNullOrEmpty()) {
+            return 0
+        }
+
+        this.total = numbers[0]
+
+        for ((i, v) in numbers.drop(1).withIndex()) {
+            this.total = calculate(this.total, v, operators[i])
+        }
+
+        return this.total
```

Comment:
아래의 코드로 개선해 보면 어떨까요? 지역 변수의 수를 최소화할 수 있어요.

```suggestion
        return numbers.reduceIndexed { index, total, number ->
            calculate(total, number, operators[index - 1])
        }
```

Reply:
오! 감사합니다 :)